armi/runLog.py

```diff
@@ -33,35 +33,45 @@
 
 """
 from __future__ import print_function
-import sys
-import os
 import collections
+
+import logging
 import operator
+import os
+import sys
 import time
 
-from six import StringIO
-
 from armi import context
 from armi.context import Mode
 from armi import meta
 
+# define some private, module-level constants
+_stderrName = "{0}.{1:04d}.stderr"
+_stdoutName = "{0}.{1:04d}.stdout"
 
-# use ordereddict so we can get right order of options in GUI.
+# NOTE: To reduce repeated calls to context later, there is a bit of ugly global
+# execution here to define what the prefixes will be in every log line.
+# NOTE: use ordereddict so we can get right order of options in GUI
+_rank = ""
+if context.MPI_RANK > 0:
+    _rank = "-{:>03d}".format(context.MPI_RANK)
 _logLevels = collections.OrderedDict(
     [
-        ("debug", (0, "dbug")),
-        ("extra", (10, "xtra")),
-        ("info", (20, "info")),
-        ("important", (25, "impt")),
-        ("prompt", (27, "prmt")),
-        ("warning", (30, "warn")),
-        ("error", (50, "err ")),
-        ("header", (100, "")),
+        ("debug", (0, "[dbug{}] ".format(_rank))),
+        ("extra", (10, "[xtra{}] ".format(_rank))),
+        ("info", (20, "[info{}] ".format(_rank))),
+        ("important", (25, "[impt{}] ".format(_rank))),
+        ("prompt", (27, "[prmt{}] ".format(_rank))),
+        ("warning", (30, "[warn{}] ".format(_rank))),
+        ("error", (50, "[err {}] ".format(_rank))),
+        ("header", (100, "".format(_rank))),
     ]
 )
+_whitespace = " " * len(max([l[1] for l in _logLevels.values()]))
 
-_stderrName = "{0}.{1:04d}.stderr"
-_stdoutName = "{0}.{1:04d}.stdout"
+# modify the logging module strings for printing
+for logValue, shortLogString in _logLevels.values():
+    logging.addLevelName(logValue, shortLogString)
```

Comment:
I will take a look at `_bootstrap`! Thanks for that.

Reply:
So, `_bootstrap.py` imports both `context.py` and `runLog.py` in their entirety.  So I don't want to be importing things from `_boostrap.py` into `runLog.py`. That's circular.

I *do* like the idea of having something less ugly like `configureLogging()`, even if all it does is fix these `logLevels` strings.

Reply:
If bootstrap imports runLog, what would be the issue with:
```python
import runLog
#...
runLog.configureLogging()
```

It seems silly/academic (and to a point, it kind of is), but it at least pushes us in the direction of "all of the infernal, unholy import-time magic happens [here]"

Reply:
I found a few issues trying to export much of the logging logic from `runLog.py` to `context.py` and `_bootstrap.py`.  It turns out we initialize our area significantly differently in runs versus in pytest.  So, there is some pytest env setup code in the repo that needs to change.

Also, some of the plugins are calling code I wanted to move out of `runLog.py` directly, so we'd have to at least keep stubs there.

I am, however, now fully convinced of Mitch's point of view that we need to move stuff out of `operatorMPI.py` and into `context.py`.  I'd like to have a conversation with him about what I found, and any specifics he has, because I think that is enough material for it's own PR.